### PR TITLE
Jeff

### PR DESCRIPTION
**Pull request by jeffreyyhuang**
Preparing for widget consolidation.
Pull request includes:
-Written tests for 70% code coverage
-Implementation of comparison modal functionality
-Additional CSS styling modifications
-Implementation of Add to Outfit functionality
-Implementation of removing an item from outfit functionality

### CODE REVIEW

File: Related/component/ComparisonModal.jsx [hunk 1]
```diff
@@ -1,52 +1,101 @@
...
+              </div>
+            </div>
+            <div className={styles.tableContainer}>
+              <table id="comparisonTable">
+                <tr className={styles.tableHeaders}>
+                  <th className={styles.leftHeader}>{this.props.currentItem[0].product.name}</th>
+                  <th className={styles.centerHeader}></th>
+                  <th className={styles.rightHeader}>{this.props.relatedItem.product.name}</th>
+                </tr>
+                {this.state.tableRows.map((feature) => (
```
Comment (kevin-the-engi) [inline Related/component/ComparisonModal.jsx:90]: Need a key here to avoid unique child error

File: Related/component/ProductInformation.jsx [hunk 2]
```diff
@@ -52,38 +48,43 @@ class ProductInformation extends React.Component {
...
-
-    hideModal() {
-      this.setState({show: false});
-    }
-
     render() {
         let displayPrice;
         if (this.state.salePrice === null) {
-            displayPrice = <div class={styles.price}>${this.state.defaultPrice}</div>;
+            displayPrice = <div class={styles.defaultPrice}>${this.state.defaultPrice}</div>;
```
Comment (kevin-the-engi) [inline Related/component/ProductInformation.jsx]:
```suggestion
            displayPrice = <div className={styles.defaultPrice}>${this.state.defaultPrice}</div>;
```
Need className not class

File: Related/component/ProductInformation.jsx [hunk 3]
```diff
@@ -52,38 +48,43 @@ class ProductInformation extends React.Component {
...
         } else {
-            displayPrice = <div class={styles.price}>${this.state.salePrice}</div>;
+            displayPrice = <div class={styles.salePrice}>${this.state.salePrice}</div>;
```
Comment (kevin-the-engi) [inline Related/component/ProductInformation.jsx]: Same as above

File: Related/component/ComparisonModal.jsx [hunk 4]
```diff
@@ -1,52 +1,101 @@
...
-            <div>
+        <div className={styles.comparisonModalDisplayBlock} onClick={this.handleClick}>
+          <section id="comparisonModal" className={styles.comparisonModalMain}>
+            <div className={styles.modalBanner}>
+              <div className={styles.modalBannerContent}>
               Comparing
+              </div>
+            </div>
+            <div className={styles.tableContainer}>
+              <table id="comparisonTable">
```
Comment (kevin-the-engi) [inline Related/component/ComparisonModal.jsx:84]: Getting warnings here. <tr> cannot appear as a child of <table>. You need to place them in <tbody>, <thead>, or <tfoot>

File: Related/component/RemoveButton.jsx [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+  let handleClick = function () {
+    props.removeFromOutfit(props.relatedItem.product_id);
+  }
+
+  return (
+    <div className={styles.removeButtonContainer} onClick={handleClick}>
+      <svg className={styles.removeButton}>
+        <circle cx="12" cy="12" r="11" stroke="black" stroke-width="2" fill="white" />
+        <path stroke="black" stroke-width="2" fill="none" d="M6.25,6.25,17.75,17.75" />
```
Comment (kevin-the-engi) [inline Related/component/RemoveButton.jsx:14]:
```suggestion
        <path stroke="black" strokeWidth="2" fill="none" d="M6.25,6.25,17.75,17.75" />
```

File: Related/component/RemoveButton.jsx [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+const RemoveButton = (props) => {
+
+  let handleClick = function () {
+    props.removeFromOutfit(props.relatedItem.product_id);
+  }
+
+  return (
+    <div className={styles.removeButtonContainer} onClick={handleClick}>
+      <svg className={styles.removeButton}>
+        <circle cx="12" cy="12" r="11" stroke="black" stroke-width="2" fill="white" />
```
Comment (kevin-the-engi) [inline Related/component/RemoveButton.jsx:13]:
```suggestion
        <circle cx="12" cy="12" r="11" stroke="black" strokeWidth="2" fill="white" />
```

File: Related/component/RemoveButton.jsx [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+        <path stroke="black" stroke-width="2" fill="none" d="M6.25,6.25,17.75,17.75" />
+        <path stroke="black" stroke-width="2" fill="none" d="M6.25,17.75,17.75,6.25" />
```
Comment (kevin-the-engi) [inline Related/component/RemoveButton.jsx:15]:
```suggestion
        <path stroke="black" strokeWidth="2" fill="none" d="M6.25,17.75,17.75,6.25" />
```

File: Related/component/ComparisonModal.jsx (hunk 4 above)
Comment (kevin-the-engi) [reply Related/component/ComparisonModal.jsx:84]: < tr> cannot appear as child of < table>, needs to be in < tbody> < tfoot> < thead>

**Review by kevin-the-engi — CHANGES_REQUESTED** [review]

File: Related/component/ProductInformation.jsx (hunk 2 above)
Comment (jeffreyyhuang) [reply Related/component/ProductInformation.jsx]: Updated to className

File: Related/component/ProductInformation.jsx (hunk 3 above)
Comment (jeffreyyhuang) [reply Related/component/ProductInformation.jsx]: updated to className

**Review by kevin-the-engi — APPROVED** [review]

File: Related/component/ComparisonModal.jsx (hunk 1 above)
Comment (jeffreyyhuang) [reply Related/component/ComparisonModal.jsx:90]: I think I removed the child error by including a key in RelatedGallery

**Pull request merged**